### DISCUSSION TITLE
[@types/google.maps] Places inconsistencies (multiple issues)

### DISCUSSION THREAD

**Discussion by spenserschwartz** [DefinitelyTyped/DefinitelyTyped#68943] [Issues with a @types package]
Hello, there are plenty of examples of google.maps mismatched types, but I will show the most basic example below: 

- [Google's Place according to Google](https://developers.google.com/maps/documentation/places/web-service/reference/rest/v1/places#Place) - one example is "editorialSummary": ([LocalizedText](https://developers.google.com/maps/documentation/places/web-service/reference/rest/v1/places#LocalizedText)). `LocalizedText` is never defined in DefinitelyTyped from what I can find either.

- [Google's Place according to @types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google.maps/index.d.ts), (line 12987) has the type as a string. It also has "editorialSummaryLanguageCode" as a key, which doesn't exist in Google's Place. 

I've confirmed Google's documents to be true (see screenshot of a response from [Google's Nearby Search (New) API](https://developers.google.com/maps/documentation/places/web-service/nearby-search#about_response). I can keep adding to this discussion with all the examples, but I have come across dozens of inconsistencies and end up making my own types to fill the gaps.

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/8887723/36262588-8c0d-45b6-8f03-bd70c760022f)

[a comment by an automated account is not shown]

**Comment by kwnevarez** [comment]
Hello, the documentation that describes the editorialSummary as LocalizedText is for the **Places API REST interface**.  And you are correct that there are no typings for LocalizedText, but that is because typings are not built for the REST interface. Typings are built for the **Maps Javascript API**, which has its own way of interfacing with the Places product that is **not** a 1 to 1 mapping of how the REST interface works. 

If you look at Maps Javascript API reference documentation for the Place object you will find [editorialSummary](https://developers.google.com/maps/documentation/javascript/reference/place#Place.editorialSummary) and [editorialSummaryLanguageCode](https://developers.google.com/maps/documentation/javascript/reference/place#Place.editorialSummaryLanguageCode) which match the [typings mentioned above](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google.maps/index.d.ts#L12987). 

I write this not having done a complete audit of all the typings so if you have examples of JS API object methods or properties that lack typings, please do share.

**Reply by spenserschwartz** [reply]
@kwnevarez I think I understand what you're saying, although it seems Google blends what is in the realm of "Places" and what is in the realm of "Maps." Is your conclusion then that DefinitelyTyped does not cover these interface types at all then?
